architecting-cloud-native-dotnet-apps-for-azure/azure-monitor.md

```diff
@@ -0,0 +1,71 @@
+
+The ultimate goal of gathering information about the state of your application is to ensure that your end users have a good experience. What better way to tell if users are experiencing issues than performing [outside-in web tests](https://docs.microsoft.com/en-us/azure/azure-monitor/app/monitor-web-app-availability)? These test can be as simple as pinging your website from locations around the world or as involved as having agents log into the site and perform actions.
+
+## Reporting data
+
+Once the data is gathered it can be manipulated, summarized and plotted into charts which allow user to instantly see when there are problems. These charts can be gathered into dashboards or into Workbooks, a multi-page report designed to tell a story about some aspect of the system.
+
+Of course, no modern application would be complete without a little bit of Artificial Intelligence or Machine Learning. To this end data [can be passed](https://www.youtube.com/watch?v=Cuza-I1g9tw) to the various Machine Learning tools in Azure to allow you to extract trends and information which would otherwise be hidden. 
+
+Application Insights provides a powerful query language called Kusto which can be used to find records, summarize them and even plot charts. For instance this query will locate all the records between 2007-11-01 and 2007-12-01, group them by state, take the top 10 and plot them as a pie chart.
+
+```
+StormEvents 
+| where StartTime >= datetime(2007-11-01) and StartTime < datetime(2007-12-01)
+| summarize count() by State
+| top 10 by count_
+| render piechart 
+```
+
+![The result of the Application Insights Query](media/application_insights_example.png)
+
+There is a [playground for experimenting with Kusto](https://dataexplorer.azure.com/clusters/help/databases/Samples) queries which is a fantastic place to spend an hour or two. Reading [sample queries](https://docs.microsoft.com/en-us/azure/kusto/query/samples) can also be instructive.
+
+## Dashboards
+
+There are several different dashboarding technologies which may be used to surface the information from Azure Monitor. Perhaps the simplest is to just run queries in Application Insights and [plot the data into a chart](https://docs.microsoft.com/en-us/azure/azure-monitor/learn/tutorial-app-dashboards). 
+
+![An example of Application Insights charts embedded in the main Azure Dashboard](media/azure_dashboard.png)
```

Comment:
Can you update all the figures to include `Figure N-M. Blah blah blah` where N = chapter number, M is sequential for the figures starting at 0, and blah is the alt text.

Example from another chapter:
```
![ Steps in the CI Pipeline](media/steps-in-the-ci-pipeline.png)
**Figure 1-13**. Steps in the CI Pipeline
```